app/uk/gov/hmrc/excisemovementcontrolsystemapi/controllers/SubmitMessageController.scala

```diff
@@ -42,9 +46,13 @@ class SubmitMessageController @Inject()(
       andThen validateErnsAction
       andThen validateLRNAction(lrn)).async(parse.xml) {
       implicit request =>
-        movementMessageConnector.submitMessage(request).flatMap {
-          case Right(_) => Future.successful(Accepted(""))
-          case Left(error) => Future.successful(error)
+        val ern = emcsUtils.getSingleErnFromMessage(request.parsedRequest.ieMessage, request.validErns)
+
+        workItemService.addWorkItemForErn(ern).flatMap { _ =>
```

Comment:
What happens here if Mongo fails (throw an exception)? Do we want catch this and show a more user friendly error message? At the moment both the workItem service and the controller do not catch a mongo error so the user will see a mongoDB error (The client already complained about this mongo error before).

Also this rise another question? Do we want this to fail if we cannot save the work item? Another choice would be to let it to carry on sending the message to EIS and when the client ask for all mssage we:
 1. go to the cache 
 2. if nothing in the cache return nothing to the use,
 3. do the polling stuff
 4. wait for the use r to GEtNewMessage again